src/coreclr/src/jit/instr.cpp

```diff
@@ -1086,8 +1089,161 @@ void CodeGen::inst_RV_TT_IV(instruction ins, emitAttr attr, regNumber reg1, GenT
+        }
+        else if (op2->isIndir() || op2->OperIsHWIntrinsic())
+        {
+            GenTree*      addr;
+            GenTreeIndir* memIndir = nullptr;
+
+            if (op2->isIndir())
+            {
+                memIndir = op2->AsIndir();
+                addr     = memIndir->Addr();
+            }
+            else
+            {
+#if defined(FEATURE_HW_INTRINSICS)
+                assert(op2->AsHWIntrinsic()->OperIsMemoryLoad());
+                assert(HWIntrinsicInfo::lookupNumArgs(op2->AsHWIntrinsic()) == 1);
+                addr = op2->gtGetOp1();
+#else
+                unreached();
 #endif // FEATURE_HW_INTRINSICS
+            }
+
+            switch (addr->OperGet())
+            {
+                case GT_LCL_VAR_ADDR:
+                {
+                    varNum = addr->AsLclVarCommon()->GetLclNum();
+                    offset = 0;
+                    break;
+                }
+
+                case GT_CLS_VAR_ADDR:
+                {
+                    GetEmitter()->emitIns_SIMD_R_R_C(ins, size, targetReg, op1Reg, addr->AsClsVar()->gtClsVarHnd, 0);
+                    return;
+                }
+
+                default:
+                {
+                    GenTreeIndir load = indirForm(op2->TypeGet(), addr);
+
+                    if (memIndir == nullptr)
+                    {
+                        // This is the HW intrinsic load case.
+                        // Until we improve the handling of addressing modes in the emitter, we'll create a
+                        // temporary GT_IND to generate code with.
+                        memIndir = &load;
+                    }
+                    GetEmitter()->emitIns_SIMD_R_R_A(ins, size, targetReg, op1Reg, memIndir);
+                    return;
+                }
+            }
+        }
+        else
+        {
+            switch (op2->OperGet())
+            {
+                case GT_LCL_FLD:
+                {
+                    varNum = op2->AsLclFld()->GetLclNum();
+                    offset = op2->AsLclFld()->GetLclOffs();
+                    break;
+                }
+
+                case GT_LCL_VAR:
+                {
+                    assert(op2->IsRegOptional() ||
+                           !compiler->lvaTable[op2->AsLclVar()->GetLclNum()].lvIsRegCandidate());
+                    varNum = op2->AsLclVar()->GetLclNum();
+                    offset = 0;
+                    break;
+                }
 
+                case GT_CNS_DBL:
+                {
+                    GenTreeDblCon*       dblCns = op2->AsDblCon();
+                    CORINFO_FIELD_HANDLE cnsDblHnd =
+                        GetEmitter()->emitFltOrDblConst(dblCns->gtDconVal, emitTypeSize(dblCns));
+                    GetEmitter()->emitIns_SIMD_R_R_C(ins, size, targetReg, op1Reg, cnsDblHnd, 0);
+                    return;
+                }
+
+                default:
+                {
+                    unreached();
+                    break;
```

Comment:
`break` is not needed after `unreached()`

Reply:
Removed.